CHANGELOG.md

```
@@ -6,6 +6,185 @@ All notable changes to this project are documented here.
+  - Comprehensive test suite with ≥85% coverage (`tool_router/core/tests/test_features.py`)
+- **Next.js Web Admin Interface**: Modern web UI for MCP Gateway management
+  - Integrated at `http://localhost:4444/admin` (Context Forge integration)
+  - Feature toggle management dashboard with 17 controls
+  - Virtual server lifecycle management (reuse existing API)
+  - Gateway status monitoring
+  - Tailwind CSS styling with dark mode support
+  - Responsive design (mobile-friendly)
+  - Production-ready JWT authentication (optional for localhost)
+  - Standalone Docker deployment with nginx reverse proxy
+
+### Added
```

markdownlint-cli2 (0.20.0)

[warning] Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)